django_forest/resources/utils/csv.py

```diff
@@ -13,19 +13,20 @@ def get_related_res(self, data, value):
         return related_res
 
     def fill_csv_relationships(self, res, record, data, params):
+        pk_name = self.Model._meta.pk.name
         for name, value in record['relationships'].items():
             related_res = self.get_related_res(data, value)
             if related_res:
-                if 'attributes' in related_res:
-                    res[name] = related_res['attributes'][params[f'fields[{name}]']]
+                if 'attributes' in related_res and pk_name in related_res['attributes']:
```

Comment:
When the table has a primary key named something other than "id", this fails because `params[f'fields[{name}]']` evaluates to "id", which doesn't exist in the "attributes" object (causing a key error).